lib/modules/apostrophe-schemas/public/js/user.js

```diff
@@ -572,6 +572,46 @@ apos.define('apostrophe-schemas', {
     });
 
     self.addFieldType({
+      name: 'object',
+      populate: function populate(data, name, $field, $el, field, callback) {
+        var $fieldset = self.findFieldset($el, name);
+        _.each(field.schema, function (subField) {
+          var $input = self.findSafeInFieldset($fieldset, subField.name, '[name=' + subField.name + ']');
+          var $subFieldSet = self.findFieldset($fieldset, subField.name);
+
+          if (subField.required === true) {
+            $subFieldSet.addClass('apos-required');
+          }
+
+          if (subField.type === 'object') {
+            // recursive call : no need for callback as the real callback will be called at the end of this loop
+            populate(data[name], subField.name, $subFieldSet, $fieldset, subField, function() { return; })
+          } else {
+            $input.val(data[name][subField.name]);
```

Comment:
Please revisit this block in general: you should be using the populate method of schemas to do this work, so that all field types work automatically, and in fact you don't have to write your own _.each loop at all.

This will eliminate my concern about your not invoking callbacks and it will also work with all field types — arrays, etc. — that are not as simple as calling val().